### PR TITLE
Fixed the issue regarding the breadcrumbs' items being too long and not displaying properly.

### PR DESCRIPTION
**Pull request by deyordanov**
Closes https://github.com/SoftUni-Internal/exam-systems-issues/issues/1379

### CODE REVIEW

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/utils/breadcrumb-utils.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    */
+    const scale = (nameLength - BreadcrumbItemNameMinLength) / (BreadcrumbItemNameMaxLength - BreadcrumbItemNameMinLength);
+    // eslint-disable-next-line max-len
+    const adjustedPercentage = BreadcrumbItemNameMinShorteningPercentage + scale * (BreadcrumbItemNameMaxShorteningPercentage - BreadcrumbItemNameMinShorteningPercentage);
+
+    return Math.round(nameLength * adjustedPercentage);
+};
+
+const trimBreadcrumbItems = (breadcrumbItems: Array<ContestBreadcrumb>): Array<ContestBreadcrumb> => {
+    // eslint-disable-next-line linebreak-style
```
Comment (gogo4ds) [inline Servers/UI/OJS.Servers.Ui/ClientApp/src/utils/breadcrumb-utils.ts:26]: Why did you disable this rule?

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/utils/breadcrumb-utils.ts (hunk 1 above)
Comment (deyordanov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/utils/breadcrumb-utils.ts]: I forgot to remove it after changing the algorithm's logic.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/utils/breadcrumb-utils.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    let breadcrumbItemsStringLength = breadcrumbItems?.reduce((acc, curr) => acc + curr.name.length, 0);
+
+    return [ ...breadcrumbItems ].reverse().reduce((acc: Array<ContestBreadcrumb>, breadcrumbItem) => {
+        const symbolCountToBeRemoved = calculateBreadcrumbNameSymbolCountToBeRemoved(breadcrumbItem.name.length);
```
[the remark this answers is not in the document]
Comment (deyordanov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/utils/breadcrumb-utils.ts]: Done.

File: Servers/UI/OJS.Servers.Ui/ClientApp/src/common/constants.ts [hunk 3]
```diff
@@ -22,6 +22,13 @@ const { OrderBy: DEFAULT_SORT_TYPE } = SortType;
 const LECTURER = 'Lecturer';
 const defaultPathIdentifier = 'api';
 
+const MaxBreadcrumbsStringLength = 100;
+const BreadcrumbItemNameMinLength = 2;
+const BreadcrumbItemNameMaxLength = 100;
+const BreadcrumbItemNameMinShorteningPercentage = 0.2;
+const BreadcrumbItemNameMaxShorteningPercentage = 0.6;
```
[the remark this answers is not in the document]
Comment (deyordanov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/common/constants.ts]: ```BreadcrumbItemNameMaxLength``` is the maximum length of the breadcrumb item's name, it is not the same as ```MaxBreadcrumbsStringLength```, which represent the sum of all breadcrumb items' names. The latter is used to define when we want to trim the breadcrumb's items.

[the remark this answers is not in the document]
Comment (deyordanov) [reply Servers/UI/OJS.Servers.Ui/ClientApp/src/common/constants.ts]: The rest of the requested changes have been implemented.

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**